src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/ConversionCompletionProvider.cs

```diff
@@ -0,0 +1,266 @@
+                    SpecialType.System_Int64,
+                    SpecialType.System_UInt64,
+                    SpecialType.System_Char,
+                    SpecialType.System_Single,
+                    SpecialType.System_Double,
+                    SpecialType.System_Decimal,
+                };
+
+                AddCompletionItemsForSpecialTypes(builder, semanticModel, container, containerIsNullable, position, convertToSpecialTypes);
+            }
+        }
+
+        private void AddCompletionItemsForSpecialTypes(ArrayBuilder<CompletionItem> builder, SemanticModel semanticModel, INamedTypeSymbol fromType, bool containerIsNullable, int position, SpecialType[] specialTypes)
+        {
+            var containerTypeName = fromType.ToMinimalDisplayString(semanticModel, position);
+            var conversionCompletionItems = from specialType in specialTypes
+                                            let targetTypeSymbol = semanticModel.Compilation.GetSpecialType(specialType)
+                                            let targetTypeName = targetTypeSymbol.ToMinimalDisplayString(semanticModel, position)
+                                            select CreateCommonCompletionItem(containerTypeName, targetTypeName, targetTypeIsNullable: containerIsNullable, position);
+            builder.AddRange(conversionCompletionItems);
+        }
+
+        private CompletionItem CreateSymbolCompletionItem(IMethodSymbol methodSymbol, string targetTypeName, bool targetTypeIsNullable, int position)
+        {
+            var optionalNullableQuestionmark = GetOptionalNullableQuestionMark(targetTypeIsNullable);
+            return SymbolCompletionItem.CreateWithSymbolId(
+                           displayTextPrefix: "(",
+                           displayText: targetTypeName,
+                           displayTextSuffix: $"{optionalNullableQuestionmark})",
+                           filterText: targetTypeName,
+                           sortText: SortText(targetTypeName),
+                           symbols: ImmutableList.Create(methodSymbol),
+                           rules: CompletionItemRules.Default,
+                           contextPosition: position,
+                           properties: CreatePropertiesBag((MinimalTypeNamePropertyName, $"{targetTypeName}{optionalNullableQuestionmark}")));
+        }
+
+        private CompletionItem CreateCommonCompletionItem(string containerTypeName, string targetTypeName, bool targetTypeIsNullable, int position)
+        {
+            var optionalNullableQuestionmark = GetOptionalNullableQuestionMark(targetTypeIsNullable);
+            return CommonCompletionItem.Create(
+                           displayTextPrefix: "(",
+                           displayText: targetTypeName,
+                           displayTextSuffix: $"{optionalNullableQuestionmark})",
+                           filterText: targetTypeName,
+                           sortText: SortText(targetTypeName),
```

Comment:
at first i thought alphabetical would be good.  But it actually does make a lot of sense to group things by their general categories.  Up to you.

Reply:
I thought about it and I think that this would do more harm than good. Suppose you have a custom type with user-defined conversions to built-in types. These would be sorted alphabetically. But conversions from built-in types would be sorted differently. The least surprise for a user is sorting the types alphabetically.